.circleci/config.yml

```diff
@@ -0,0 +1,158 @@
+          key: v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - .
+            - ~/.ssh/known_hosts
+
+  dependencies: &dependencies
+    working_directory: ~/country-template
+    docker:
+      - image: python:2.7.14
+    environment:
+      CREATE_VENV: "virtualenv"
+
+    steps:
+      - restore_cache:
+          keys:
+            - v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Create virtualenv
+          command: |
+            mkdir -p /tmp/venv/country-template
+            ${CREATE_VENV} /tmp/venv/country-template
+
+      # pip >= 8.0 needed to be compatible with "manylinux" wheels, used by numpy >= 1.11
+      - run:
+          name: Install dependencies
+          command: |
+            . /tmp/venv/country-template/bin/activate
+            pip install --upgrade pip twine wheel
+            pip install .[test] --upgrade
+
+      # Uncomment and adapt the next line to use a particular feature branch of OpenFisca-Core to run Circle CI tests
+      # - run:
+      #     name: Install a particular feature branch of OpenFisca-Core
+      #     command: |
+      #       . /tmp/venv/country-template/bin/activate
+      #       pip install --editable git+https://github.com/openfisca/openfisca-core.git@BRANCH_NAME#egg=OpenFisca-Core
+
+      - save_cache:
+          # PY_VERSION is set in the context
+          key: v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - /tmp/venv/country-template
+
+  dependencies_python3:
+    <<: *dependencies
+    docker:
+      - image: python:3.6
+    environment:
+      CREATE_VENV: "python -m venv"
+
+  tests: &tests
+    working_directory: ~/country-template
+    docker:
+      - image: python:2.7.14
+
+    steps:
+      - restore_cache:
+          keys:
+            - v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Run tests
+          command: |
+            . /tmp/venv/country-template/bin/activate
+            make test
+            . check-version-bump.sh
+
+  tests_python3:
+    <<: *tests
+    docker:
+      - image: python:3.6
+
+  deploy: &deploy
+    working_directory: ~/country-template
+    docker:
+      - image: python:2.7.14
+    environment:
+      PYPI_USERNAME: openfisca-bot
+      # PYPI_PASSWORD: this value is set in CircleCI's web interface; do not set it here, it is a secret!
+
+    steps:
+      - restore_cache:
+          keys:
+            - v1-checkout-{{ .Environment.CIRCLE_SHA1 }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ .Environment.PY_VERSION }}-dependencies-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Deploy (if version bump)
+          command: |
+            . /tmp/venv/country-template/bin/activate
+            . deploy-if-version-bump.sh
```

Comment:
Unless CircleCI does something special, the `.` is a non-readable equivalent of `source`. I recommend we use `source` over `.` for readability. I also believe that `. deploy-if-version-bump.sh` should not be sourced, but directly executed, and that it currently works only as a side-effect of sourcing.

Reply:
Done